lib/Driver/UnixToolChains.cpp

```diff
@@ -205,6 +200,11 @@ toolchains::GenericUnix::constructInvocation(const DynamicLinkJobAction &job,
     staticStdlib = true;
   }
 
+  if (getTriple().getOS() == llvm::Triple::Linux &&
```

Comment:
Is there a reason for the Linux check?  It should be applicable for any ELFish target

Reply:
Lets get it working for one target first. It should be applicable to others but without any testing/ CI its hard to confirm. It should be easier to expand to others later.